### ISSUE TITLE
Can't access external website with local Kubernetes in docker-desktop 4.25.0 releae

### ISSUE THREAD

**Issue by cdlliuy** [docker/for-win#13768]
### Description

with the latest https://docs.docker.com/desktop/release-notes/#4250 4.25.0 release,  our application running on the local Kubernetes can't access external website anymore. 

The coredns pods kept failing with 
```
[ERROR] plugin/errors: 2 xxx. AAAA: dns: overflowing header size
[ERROR] plugin/errors: 2 xxx A: dns: overflow unpacking uint16
```

We already identified it is a coredns issue with https://github.com/coredns/coredns/issues/5998,   and can be migrated by downgrade the coredns version to v1.10.0

can the upstream docker-desktop fix the coredns version issue asap with a new release?   

### Reproduce

* deploy any container to local kuberenets hosted by docker
* do curl cmd inside the container: 
```
curl https://login.microsoftonline.com -v
* Could not resolve host: login.microsoftonline.com
* Closing connection 0
curl: (6) Could not resolve host: login.microsoftonline.com
```

*  Check coredns logs: 
[ERROR] plugin/errors: 2 login.microsoftonline.com. A: dns: overflow unpacking uint16
[ERROR] plugin/errors: 2 login.microsoftonline.com. AAAA: dns: overflowing header size

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:32:48 2023
 OS/Arch:           windows/amd64
 Context:           default

Server: Docker Desktop 4.25.0 (126437)
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       1a79695
  Built:            Mon Sep  4 12:32:16 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.22
  GitCommit:        8165feabfdfe38c65b599c4993d227328c231fca
 runc:
  Version:          1.1.8
  GitCommit:        v1.1.8-0-g82f18fe
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
N/A
```

### Diagnostics ID

N/A

**Comment by JoFrMueller** [comment]
For us it's similar though doesn't even have to be K8s.

```
/ # curl https://api.nuget.org/ -v
* processing: https://api.nuget.org/
* Could not resolve host: api.nuget.org
* Closing connection
curl: (6) Could not resolve host: api.nuget.org
```

Interestingly enough many others like google work:

```
/ # curl https://google.com/ -v
* processing: https://google.com/
*   Trying 142.251.36.174:443...
* Connected to google.com (142.251.36.174) port 443
* ALPN: offers h2,http/1.1
```

Going back to 4.24.4 all is good again...

**Comment by aarondonohue** [comment]
I too have encountered this issue. Downgrading from 4.26.1 to 4.24.2 solved my problem.

**Comment by rsheasby** [comment]
Same issue. Using a 2 month old version of Docker Desktop is sub-optimal, has anyone found a workaround?

**Comment by PhilHamlin-MS** [comment]
That issue is caused by this issue<https://github.com/coredns/coredns/issues/5998>, so our workaround was a DNS solution not involving CoreDNS for now.
________________________________

**Comment by jasase** [comment]
Does some one know if this problem is solved in Version 4.27.0?

**Comment by PhilHamlin-MS** [comment]
The GitHub issue's still open.  (Apologies, I can't tell whether that's you asking exactly this question on GitHub.)
Can't access external website with local Kubernetes in docker-desktop 4.25.0 releae · Issue #13768 · docker/for-win (github.com)<https://github.com/docker/for-win/issues/13768>
________________________________

**Comment by pcatlin** [comment]
No, still broken for me using 4.27.0 - seems like all OS's are affected too - [Mac #7110](https://github.com/docker/for-mac/issues/7110)

**Comment by dennis-yemelyanov** [comment]
any plans to fix?

**Comment by crowne** [comment]
As @cdlliuy says, this can be resolved by rolling coredns back from v1.10.1 to v1.10.0

here's the patch command:

```shell
kubectl patch deployment coredns -n kube-system -p '{"spec":{"template":{"spec":{"containers":[{"name":"coredns","image":"registry.k8s.io/coredns/coredns:v1.10.0"}]}}}}'
```

**Comment by bogdanghervan** [comment]
Still broken in 4.28.0.

**Comment by qcaas-nhs-sjt** [comment]
I've tried rolling back to version 1.10.0 on docker desktop, however this issue is still occurring for me. When can we expect a proper fix for this?

**Comment by JoFrMueller** [comment]
@qcaas-nhs-sjt - we run on Docker Desktop `4.24.2` like @aarondonohue mentioned since ages felt, works very well for us. Just never upgrade - juniors will rage, but we had 10 erroneous upgrade attempts, so we just don't care anymore for and choose that version only. It's only dev machines anyways in our case, - so we're lucky there.

Ensure to have uninstalled Docker Desktop properly before installing the old version, then all should be fine - at least for `docker compose`, K8s I don't have any clue - sorry.

**Comment by qcaas-nhs-sjt** [comment]
> @qcaas-nhs-sjt - we run on Docker Desktop `4.24.2` like @aarondonohue mentioned since ages felt, works very well for us. Just never upgrade - juniors will rage, but we had 10 erroneous upgrade attempts, so we just don't care anymore for and choose that version only. It's only dev machines anyways in our case, - so we're lucky there.
> 
> Ensure to have uninstalled Docker Desktop properly before installing the old version, then all should be fine - at least for `docker compose`, K8s I don't have any clue - sorry.

Thanks for this, I understand that this is a suggested workaround but we are now on v4.28.0, I have a team of people that are new to kubernetes that I am building a framework for and many of these new developers will be using windows machines in a windows environment where the centrally managed IT team will be wanting to ensure the environment is properly patched for security issues. As a result we must provide an environment that is updateable. There are currently security fixes in the patches since so it is really not suitable in our environment so a fix for this is needed and while we can work around it in this way we really shouldn't be.

**Comment by jasase** [comment]
any plans to fix? Seems there is no progress on this topic.

**Comment by KavyaShree25** [comment]
i m unable to install previous versions of docker desktop.any help here

**Comment by patrickhuber** [comment]
I managed to use a coredns config map to work around this issue. Sending all requests to the login.microsoftonline.com domain to cloudflare 1.1.1.1

> coredns.yml

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: coredns
  namespace: kube-system
data:
  Corefile: |
    login.microsoftonline.com {
      forward . 1.1.1.1
    }

    .:53 {
        errors
        health {
           lameduck 5s
        }
        ready
        kubernetes cluster.local in-addr.arpa ip6.arpa {
           pods insecure
           fallthrough in-addr.arpa ip6.arpa
           ttl 30
        }
        prometheus :9153
        forward . /etc/resolv.conf {
           max_concurrent 1000
        }
        cache 30
        loop
        reload
        loadbalance
    }
```

```bash
kubectl apply -f coredns.yml
```

Here is the documentation on patching dns 
* https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/#are-dns-queries-being-received-processed

Here is the documentation on coredns forward entries
* https://coredns.io/plugins/forward/

I exported my config using the following command before adding in the `forward` section on `login.microsoftonline.com`

```bash
kubectl -n kube-system get configmap coredns
```

It appears coredns 1.11.3 fixes a UDP DNS Overflow but it isn't available to downstream kubernetes as a container yet so this is temporary until the new image is available. https://github.com/coredns/coredns/releases/tag/v1.11.3

You will get a warning about missing annotations. I think this is because I didn't edit the config, I exported it and re-imported it. kubectl fixes it so I guess its not too much of an issue.
